### PR TITLE
DDF-3550 Improved error messages for ambiguous unknown errors

### PR DESCRIPTION
**Pull request by mackncheesiest**
#### What does this PR do?
Adds more informative error messages in Intrigue for specific situations where "Unknown Error" appeared otherwise.

#### Who is reviewing it? 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)
@andrewkfiedler @adimka @mojogitoverhere @Variadicism @brianfelix 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams

#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@bdeining
@millerw8

#### How should this be tested? (List steps with links to updated documentation)
Build and install DDF, navigate to Intrigue, create a query to put the workspace in a "modified" state, turn off DDF, and
(i) attempt to save that query. You should receive an "Error Saving Workspace" message
(ii) wait for source polling to occur. You should receive an "Error Polling Sources" message

#### What are the relevant tickets?
[DDF-3550](https://codice.atlassian.net/browse/DDF-3550)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by mojogitoverhere** [comment]
I like the idea behind these changes but I'm not too fond of listing the possible errors in `ApplicationSetup.js`. Can we move these error messages to where their respective ajax calls are made and use custom error handlers to call `announcement.announce(...)` instead?

**Comment by mackncheesiest** [comment]
Some of them (like the source polling error) can't be moved without a bit more work because the library gets in the way of passing in a custom error handler.

I originally wanted to move the error message for the source polling, for example, into a custom handler in `Sources.js`, but the `poller` library we use to [implement](https://github.com/codice/ddf/blob/ae2094c7331cf71b52b17387dd85744a4f6c55cc/catalog/ui/catalog-ui-search/src/main/webapp/js/model/Sources.js#L111-L115) the source polling doesn't actually do anything with properties that are passed in to it.

I glanced at the [source](https://github.com/uzikilon/backbone-poller/blob/master/backbone.poller.js#L120) for poller and noticed that it accepted an options parameter that I assumed might pass through to the jquery ajax request, but after messing with it for a while, I realized that they don't actually update/extend `this.options` with `options` ever, so I don't think it can be done without modifying that library or refactoring source polling to use our own logic.

**Comment by mackncheesiest** [comment]
I thought I tried passing it in to both the `poller.get` and `poller.start` calls, but looking at `poller.get` again, it looks like that might be more feasible. I can give it another shot

File: catalog/ui/catalog-ui-search/src/main/webapp/js/ApplicationSetup.js [hunk 1]
```diff
@@ -64,13 +64,24 @@ require([
...
 
     let getErrorResponse = function (event, jqxhr, settings, throwError) {
-        switch (jqxhr.status) {
-            case 403:
-                return {title: 'Forbidden', message: 'Not Authorized'};
-            case 405:
-                return {title: 'Error', message: 'Method not allowed. Please try refreshing your browser'};
-            default:
-                return {title: 'Error', message: 'Unknown Error' + getShortErrorMessage(throwError)};
+        if (jqxhr.getResponseHeader('content-type') === 'application/json' && jqxhr.responseText.startsWith('<') &&
```
Comment (adimka) [inline catalog/ui/catalog-ui-search/src/main/webapp/js/ApplicationSetup.js]: I wouldn't mind factoring these long booleans out into descriptively named methods. 😺

**Comment by bdeining** [comment]
build now

**Pull request merged**